### PR TITLE
Change playground-code-editor to prismjs

### PR DESCRIPTION
**Pull request by GeorgePlukov**
Switches to using the PrismJs code highlighting and adds css to theme according to jeffs designs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "markdown-it-modify-token": "^1",
     "mocha": "^9",
     "playground-elements": "^0.9",
+    "prismjs": "^1.23",
```
Comment (margaree) [inline package.json:45]: This can just be simplified to `^1`

File: pages/assets/code-style.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+/* stylelint-disable */
+import { css } from 'lit-element';
+
+export const themeStyles = css`
+	code,
+	pre {
+		background: none;
+		/* this color is for base text within code snippets */
+		color: var(--d2l-color-gypsum);
+		font-family: 'Lato', 'Lucida Sans Unicode', 'Lucida Grande', sans-serif;
```
Comment (margaree) [inline pages/assets/code-style.js:10]: In core the demos use `font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;` and just personally I like the look a bit better for code.

File: pages/assets/demo-snippet.js [hunk 3]
```diff
@@ -1,10 +1,13 @@
+import 'prismjs/prism.js';
```
Comment (margaree) [inline pages/assets/demo-snippet.js:1]: Should this come after `playground-elements/playground-code-editor`?

File: pages/assets/demo-snippet.js [hunk 4]
```diff
@@ -40,38 +43,36 @@ class ComponentCatalogDemoSnippetWrapper extends LitElement {
 			*/
 			hideCode: { type: Boolean, attribute: 'hide-code' },
 			/**
-			* Should the attribute table be rendered for interactivity
+			* Hide the IFramed demo
+			*/
+			hideDemo: { type: Boolean, attribute: 'code-only' },
+			/**
+			* Should the attribute table be rendered for interactiv	ity
```
Comment (margaree) [inline pages/assets/demo-snippet.js:50]: A space got added in `interactivity`

File: pages/assets/demo-snippet.js [hunk 5]
```diff
@@ -40,38 +43,36 @@ class ComponentCatalogDemoSnippetWrapper extends LitElement {
...
+			/**
 			* Is the preview resizable
 			*/
 			resizable : { type: Boolean, reflect: true },
 			_attributes: { type: Object, reflect: true }
 		};
 	}
 	static get styles() {
-		return css`
+		return [css`
```
Comment (margaree) [inline pages/assets/demo-snippet.js:65]: Normally we'd have `themeStyles` first before the in-file css

File: pages/assets/demo-snippet.js [hunk 6]
```diff
@@ -84,27 +85,28 @@ class ComponentCatalogDemoSnippetWrapper extends LitElement {
...
 	}
 
 	constructor() {
 		super();
 		this._attributes = {};
 		this.hideCode = false;
+		this.hideDemo = false;
 		this.interactive = false;
 		this.resizable = false;
+		this.language = 'html';
```
Comment (margaree) [inline pages/assets/demo-snippet.js:98]: nit: alphabetical order

File: pages/assets/code-style.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+		word-spacing: normal;
+		word-wrap: normal;
+	}
+
+	/* Code blocks */
+	pre {
+		border-radius: 0 0 10px 10px;
+		margin: 0;
+		overflow: auto;
+		padding: 40px 24px 24px 40px;
```
Comment (margaree) [inline pages/assets/code-style.js:31]: I checked out the demo site and I think this is creating some uneven top and bottom padding (not sure if that's expected/desired)

File: pages/assets/demo-snippet.js [hunk 8]
```diff
@@ -146,20 +148,37 @@ class ComponentCatalogDemoSnippetWrapper extends LitElement {
...
+		this._highlightedCodeSnippet = Prism.highlight(this.code, Prism.languages[this.language], this.language);
+	}
+
 	render() {
 		const codeSnippet = this.code;
 		return html`
-			<d2l-component-catalog-demo-resizable-preview code=${codeSnippet} imports=${this.imports} ?resizable=${this.resizable} ?attached=${!this.hideCode} size=${ifDefined(this.size)}></d2l-component-catalog-demo-resizable-preview>
+			${ !this.hideDemo ? html`
+				<d2l-component-catalog-demo-resizable-preview
+					code=${codeSnippet}
```
Comment (margaree) [inline pages/assets/demo-snippet.js:161]: nit: I think we usually try to alphabetize these attributes

File: pages/assets/code-style.js (hunk 2 above)
Comment (GeorgePlukov) [reply pages/assets/code-style.js:10]: I pulled the fonts from the XD, I'll reach out to Jeff and see what the preferred font is

Comment (GeorgePlukov) [reply pages/assets/code-style.js]: Jeff agreed that the monospace font is nicer for code. I switched it over and changed a bit of the font size/styling to make it fit better with the rest of the page

**Review by margaree — APPROVED** [review]
Looks good!

File: pages/assets/code-style.js [hunk 9]
```diff
@@ -0,0 +1,93 @@
+/* stylelint-disable */
```
Comment (dbatiste) [inline pages/assets/code-style.js:1]: FWIW, we have a couple of prismjs stylesheets in core.  I think I got one of them that tries to model after VS Code dark.  🤔

**Pull request merged**